protocol/x/affiliates/keeper/keeper_test.go

```diff
@@ -14,3 +20,193 @@ func TestLogger(t *testing.T) {
 	logger := tApp.App.AffiliatesKeeper.Logger(ctx)
 	require.NotNil(t, logger)
 }
+
+func TestRegisterAffiliate(t *testing.T) {
+	tApp := testapp.NewTestAppBuilder(t).Build()
+	ctx := tApp.InitChain()
+	k := tApp.App.AffiliatesKeeper
+
+	tests := []struct {
+		name        string
+		referee     string
+		affiliate   string
+		expectError bool
+	}{
+		{
+			name:        "Register new affiliate",
+			referee:     "referee1",
+			affiliate:   "affiliate1",
+			expectError: false,
+		},
+		{
+			name:        "Register existing referee",
```

Comment:
Test cases should be independent instead of accumulative; otherwise it'll be hard to add more test cases. Considering adding a `setup` function as test case field, that takes a pointer to `AffiliatesKeeper` and conduct relevant setup (e.g. registering a pre-existing affilaite.